src/main/java/org/weasis/dicom/google/api/GoogleAPIClient.java

```diff
@@ -319,40 +319,32 @@ public static String getImageUrl(DicomStore store, String studyId) {
     }
 
     private String formatQuery(StudyQuery query) {
-        String allItems = "?includefield=all";
-        if (query == null) {
-            return allItems;
-        }
-
         List<String> parameters = new ArrayList<>();
-        if (isNotBlank(query.getPatientName())) {
-            parameters.add("PatientName=" + urlEncode(query.getPatientName()));
-        }
-
-        if (isNotBlank(query.getPatientId())) {
-            parameters.add("PatientID=" + urlEncode(query.getPatientId()));
-        }
-
-        if (isNotBlank(query.getAccessionNumber())) {
-            parameters.add("AccessionNumber=" + urlEncode(query.getAccessionNumber()));
-        }
-
-        if (query.getStartDate() != null && query.getEndDate() != null) {
-            parameters.add("StudyDate="
-                    + urlEncode(DATE_FORMAT.format(query.getStartDate()))
-                    + "-" + urlEncode(DATE_FORMAT.format(query.getEndDate()))
-            );
-        }
 
-        if (isNotBlank(query.getPhysicianName())) {
-            parameters.add("ReferringPhysicianName=" + urlEncode(query.getPhysicianName()));
-        }
-
-        if (parameters.isEmpty()) {
-            return allItems;
+        if (Objects.nonNull(query)) {
+            if (isNotBlank(query.getPatientName())) {
+                parameters.add("PatientName=" + urlEncode(query.getPatientName()));
+                parameters.add("fuzzymatching=true");
+            }
+            if (isNotBlank(query.getPatientId())) {
+                parameters.add("PatientID=" + urlEncode(query.getPatientId()));
+            }
+            if (isNotBlank(query.getAccessionNumber())) {
+                parameters.add("AccessionNumber=" + urlEncode(query.getAccessionNumber()));
+            }
+            if (query.getStartDate() != null && query.getEndDate() != null) {
+                parameters.add("StudyDate="
+                        + urlEncode(DATE_FORMAT.format(query.getStartDate()))
+                        + "-" + urlEncode(DATE_FORMAT.format(query.getEndDate()))
+                );
+            }
+            if (isNotBlank(query.getPhysicianName())) {
+                parameters.add("ReferringPhysicianName=" + urlEncode(query.getPhysicianName()));
+            }
         } else {
-            return "?" + join(parameters, "&");
+            parameters.add("includefield=all");
         }
-    }
 
+        return "?" + join(parameters, "&");
```

Comment:
Does join handle the case where parameters are empty skips adding any '&'?

Reply:
it does, logic of this join quite straight forward


    public static String join(Collection<String> collection, String joinString) {
        StringBuilder builder = new StringBuilder();
        for (String str : collection) {
            if (builder.length() > 0) {
                builder.append(joinString);
            }
            builder.append(str);
        }
        return builder.toString();
    }

Reply:
cool thanks.